controller/internal/routes/identity_api_model.go

```diff
@@ -34,357 +36,246 @@ const (
-	result := &model.Identity{}
-	result.Id = id
-	result.Name = stringz.OrEmpty(i.Name)
-	result.IsAdmin = boolOrFalse(i.IsAdmin)
-	result.IsDefaultAdmin = false
-	result.IdentityTypeId = stringz.OrEmpty(i.Type)
-	result.RoleAttributes = i.RoleAttributes
-	result.Tags = i.Tags
-	return result
+type IdentityLinkFactoryImpl struct {
+	BasicLinkFactory
 }
 
-type IdentityApiCreate struct {
-	Tags           map[string]interface{} `json:"tags"`
-	Enrollment     map[string]interface{} `json:"enrollment"`
-	Name           *string                `json:"name"`
-	Type           *string                `json:"type"`
-	IsAdmin        *bool                  `json:"isAdmin"`
-	RoleAttributes []string               `json:"roleAttributes"`
-}
+func (factory *IdentityLinkFactoryImpl) Links(entity models.Entity) rest_model.Links {
+	links := factory.BasicLinkFactory.Links(entity)
+	links[EntityNameEdgeRouterPolicy] = factory.NewNestedLink(entity, EntityNameEdgeRouter)
+	links[EntityNameEdgeRouterPolicy] = factory.NewNestedLink(entity, EntityNameEdgeRouter)
 
-func NewIdentityApiCreate() *IdentityApiCreate {
-	f := false
-	return &IdentityApiCreate{
-		IsAdmin: &f,
-	}
+	return links
 }
 
-func (i *IdentityApiCreate) ToModel() (*model.Identity, []*model.Enrollment) {
-	identity := &model.Identity{}
-	identity.Name = stringz.OrEmpty(i.Name)
-	identity.IdentityTypeId = stringz.OrEmpty(i.Type)
-	identity.IsDefaultAdmin = false
-	identity.IsAdmin = boolOrFalse(i.IsAdmin)
-	identity.RoleAttributes = i.RoleAttributes
-	identity.Tags = i.Tags
-
+func MapCreateIdentityToModel(identity *rest_model.IdentityCreate, identityTypeId string) (*model.Identity, []*model.Enrollment) {
 	var enrollments []*model.Enrollment
-	if caId, ok := i.Enrollment[persistence.MethodEnrollOttCa]; ok {
-		caId := caId.(string)
-		enrollment := &model.Enrollment{
-			Method: persistence.MethodEnrollOttCa,
-			CaId:   &caId,
-		}
-		enrollments = append(enrollments, enrollment)
-	}
 
-	if _, ok := i.Enrollment[persistence.MethodEnrollOtt]; ok {
-		enrollments = append(enrollments, &model.Enrollment{
-			Method: persistence.MethodEnrollOtt,
-		})
+	ret := &model.Identity{
+		BaseEntity: models.BaseEntity{
+			Tags: identity.Tags,
+		},
+		Name:           stringz.OrEmpty(identity.Name),
+		IdentityTypeId: identityTypeId,
+		IsDefaultAdmin: false,
+		IsAdmin:        *identity.IsAdmin,
+		RoleAttributes: identity.RoleAttributes,
 	}
 
-	if val, ok := i.Enrollment[persistence.MethodEnrollUpdb]; ok {
-		username := val.(string)
-		enrollments = append(enrollments, &model.Enrollment{
-			Method:   persistence.MethodEnrollUpdb,
-			Username: &username,
-		})
+	if identity.Enrollment != nil {
+		if identity.Enrollment.Ott {
+			enrollments = append(enrollments, &model.Enrollment{
+				BaseEntity: models.BaseEntity{},
+				Method:     persistence.MethodEnrollOtt,
+				Token:      uuid.New().String(),
+			})
+		} else if identity.Enrollment.Ottca != "" {
```

Comment:
i always always always trim the input "just in case". probably wouldn't hurt to trim `identity.Enrollment.Ottca` and `identity.Enrollment.Updb`?

Reply:
It isn't the APIs responsibility to trim input. The `Ottca` value is validated later by looking up the CA.